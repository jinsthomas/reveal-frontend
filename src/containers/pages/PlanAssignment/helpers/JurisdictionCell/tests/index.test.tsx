import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { openSRPJurisdiction } from '../../JurisdictionAssignmentForm/tests/fixtures';
import { JurisdictionCell } from '../index';

describe('PlanAssignment/JurisdictionCell', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders leaf nodes', () => {
    const wrapper = mount(
      <MemoryRouter>
        <JurisdictionCell
          node={openSRPJurisdiction}
          jurisdictionTree={[]}
          url={`/${openSRPJurisdiction.id}`}
        />
      </MemoryRouter>
    );
    expect(toJson(wrapper.find('span'))).toMatchSnapshot('Leaf node');
    wrapper.unmount();
  });

  it('renders leaf non-leaf nodes', () => {
    const wrapper = mount(
      <MemoryRouter>
        <JurisdictionCell
          node={openSRPJurisdiction}
          jurisdictionTree={[
            { jurisdiction_id: '12409', jurisdiction_parent_id: openSRPJurisdiction.id },
          ]}
          url={`/${openSRPJurisdiction.id}`}
        />
      </MemoryRouter>
    );
    expect(toJson(wrapper.find('Link'))).toMatchSnapshot('Non-Leaf node');
    wrapper.unmount();
  });
});
