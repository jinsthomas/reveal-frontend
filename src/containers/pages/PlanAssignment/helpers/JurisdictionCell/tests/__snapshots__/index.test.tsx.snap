// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanAssignment/JurisdictionCell renders leaf nodes: Leaf node 1`] = `
<span
  key="fca0d71d-0410-45d3-8305-a9f092a150b8-span"
>
  ra_ksh_2
</span>
`;

exports[`PlanAssignment/JurisdictionCell renders leaf non-leaf nodes: Non-Leaf node 1`] = `
<Link
  key="cec79f21-33c3-43f5-a8af-59a47aa61b84-link"
  replace={false}
  to="/cec79f21-33c3-43f5-a8af-59a47aa61b84"
>
  <a
    href="/cec79f21-33c3-43f5-a8af-59a47aa61b84"
    onClick={[Function]}
  >
    ra Luapula
  </a>
</Link>
`;
